[Java] Fix out-dated signatures of JNI methods

jovan-wong opened the pull request:
<!--
Thank you for your contribution!

Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request.
-->

## What do these changes do?
When Java code invoke native code by JNI, it will find the function by the function name which is declared in native code(`org_ray_spi_impl_DefaultLocalSchedulerClient.cc`). If there are some overloading functions, the JNI will find the most matching one by the `signature` in the comment, like this:
```c++
/*
 * Class:     org_ray_spi_impl_DefaultLocalSchedulerClient
 * Method:    _reconstructObjects
 * Signature: (J[BZ)V   /******This is the Signature declaration.******/
 */
```
Many signatures are not correct yet. 
But because there are no overloading, this issue is not triggered.

In this PR, I do these things:
1) Renamed the native JNI methods and some parameters of JNI methods. 
2) Fixed native JNI methods' signatures by `javah` tool.
3) Removed some useless native methods.

<!-- Please give a short brief about these changes. -->

## Related issue number
N/A

<!-- Are there any issues opened that will be resolved by merging this change? -->

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/7822/
Test FAILed.

jovan-wong commented:
I have updated some code.

In conclusion, it contains 3 changes:
1) Renamed the native JNI methods and some parameters of JNI methods. 
2) Fixed native JNI methods' signatures by `javah` tool.
3) Removed some useless native methods.

@robertnishihara  @raulchen  Could u plz review this again?

jovan-wong commented:
@eric-jj cc

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/7852/
Test PASSed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/7856/
Test FAILed.

jovan-wong commented:
I didn't change any py file, and just changed `org_ray_spi_impl_DefaultLocalSchedulerClient.h` and `org_ray_spi_impl_DefaultLocalSchedulerClient.cc` which are used for Java. 
But in many times, the CI job `#1` and `#2` can't pass.

```shell
=============== 1 failed, 47 passed, 8 skipped in 218.85 seconds ===============
The command "python -m pytest -v test/actor_test.py" exited with 1.
```
@robertnishihara  Do u guys have any idea about it?

robertnishihara commented on src/local_scheduler/lib/java/org_ray_spi_impl_DefaultLocalSchedulerClient.cc:
```diff
@@ -153,7 +153,8 @@ Java_org_ray_spi_impl_DefaultLocalSchedulerClient_nativeReconstructObjects(
   for (int i = 0; i < len; i++) {
-    jbyteArray object_id_bytes = (jbyteArray) env->GetObjectArrayElement(objectIds, i);
+    jbyteArray object_id_bytes =
+        (jbyteArray) env->GetObjectArrayElement(objectIds, i);
```
We should not use C style casts here

robertnishihara commented on src/local_scheduler/lib/java/org_ray_spi_impl_DefaultLocalSchedulerClient.cc:
```diff
@@ -209,7 +211,8 @@ Java_org_ray_spi_impl_DefaultLocalSchedulerClient_nativeWaitObject(
   for (int i = 0; i < len; i++) {
-    jbyteArray object_id_bytes = (jbyteArray) env->GetObjectArrayElement(objectIds, i);
+    jbyteArray object_id_bytes =
+        (jbyteArray) env->GetObjectArrayElement(objectIds, i);
```
C style cast

robertnishihara commented:
@jovan-wong I think those test failures are unrelated. Can you remove the C-style casts? Then I think we can merge it.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/7883/
Test PASSed.
